libraries/chain/transaction_context.cpp

```diff
@@ -128,6 +128,26 @@ namespace eosio { namespace chain {
             }
          }
       }
+
+      if( control.is_builtin_activated( builtin_protocol_feature_t::register_transaction_hook ) ) {
+         auto gprops = control.get_global_properties();
+         const uint32_t preexecution_hook_type = 0;
+
+         auto it = find_if(std::begin(gprops.transaction_hooks), std::end(gprops.transaction_hooks),
+                           [&](transaction_hook &t_hook) {
+                              return t_hook.type == preexecution_hook_type;
+                           }
+         );
+
+         if (it != std::end(gprops.transaction_hooks)) {
+            chain::action preexecution_hook_action{{}, it->contract, it->action, {}};
+            int action_ordinal = schedule_action(preexecution_hook_action, preexecution_hook_action.account, false, 0, 0);
+            apply_context acontext( control, *this, action_ordinal, 0 );
+            acontext.exec();
+            trace->action_traces.pop_back();
```

Comment:
Going to all the expense of creating an action_trace only to throw it away does not seem very efficient.

Reply:
I added it back